### PR TITLE
Added support for a bunch of popular alternative search engines

### PR DESCRIPTION
**Pull request by vankessel**
Support for ecosia, brave, swisscows, metager, mojeek, and two top searx instances.

### CODE REVIEW

**Comment by psidex** [comment]
Hi! Thanks for the PR, I'll take a look and try and get it merged in soon 😄

File: src/scripts/main.ts [hunk 1]
```diff
@@ -28,15 +28,27 @@ function processRequest(r: WebRequest.OnBeforeRequestDetailsType): WebRequest.Bl
 
   // Get query text or empty string
   const url = new URL(r.url);
-  let queryText = url.searchParams.get('q')?.trim() ?? '';
+  let queryParam = url.searchParams.get('q');
+  if (queryParam === null) {
+    queryParam = url.searchParams.get('query');
+  } else if (queryParam === null) {
```
Comment (psidex) [inline src/scripts/main.ts:34]: After some brief testing it seems this should be

```suggestion
  }
  if (queryParam === null) {
```

File: src/scripts/main.ts [hunk 2]
```diff
@@ -113,6 +125,13 @@ function main(): void {
         '*://*.duckduckgo.com/*',
         '*://*.qwant.com/*',
         '*://*.startpage.com/*',
+        '*://*.ecosia.org/*',
+        '*://*.brave.com/*',
+        '*://*.swisscows.com/*',
```
Comment (psidex) [inline src/scripts/main.ts:130]: swisscows doesn't seem to work, please investigate or remove this domain from the PR

**Review by psidex — CHANGES_REQUESTED** [review]
Looks good other than that! I would appreciate if you could update the readme line that says "This currently only works if you have Google, [etc]" to something that represents the engines it supports now 😄

**Comment by vankessel** [comment]
Thanks for the review. I had tested it from the search bar not the homepages so I missed the swisscows POST case. I'll fix it up after Christmas. Happy holidays and new year!

**Comment by psidex** [comment]
Ah yes I did the opposite test! Yeah no worries, take your time :) Happy holidays!

**Comment by vankessel** [comment]
I've removed Swisscows since I couldn't get it to work. Each request from the homepage does multiple requests of type 'xmlhttprequest' which gets blocked by the early return `r.type !== 'main_frame'`, and it's not as simple as allowing those requests past that check.

**Review by psidex — APPROVED** [review]

**Comment by psidex** [comment]
LGTM, Thanks! I'll try to get an update pushed to ff / chrome stores this week

**Comment by psidex** [comment]
Sorry for the wait @vankessel @i-report-bugs, I've just submitted 0.9.5 to the Firefox & Chrome stores, should be available soon!

**Pull request merged**